### PR TITLE
Add tests for RichTable

### PR DESCRIPTION
**Pull request by tyxchen**
This should hopefully resolve some of our coverage issues.

Also improves coverage for Visor, Loader, and shared/hooks.

### CODE REVIEW

**Comment by tyxchen** [comment]
@vincentmacri So it looks like the negative changes in coverage might be something to live with as we slowly add more and more tests and clean up/fix code in doing so, since many of the lines that changed to being marked as untested were never properly tested to begin with.

**Comment by tyxchen** [comment]
@vincentmacri Also, nice job on the size-limit action PR --- not being spammed with comments with every push is a huge improvement :smiley:

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -48,4 +48,4 @@ jobs:
     - uses: codecov/codecov-action@v1
       if: matrix.os == 'ubuntu-16.04'
       with:
-        file: coverage/coverage-final.json
+        file: coverage/lcov.info
```
Comment (vincentmacri) [inline .github/workflows/test.yml:51]: Why this change?

File: package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "jest": {
     "coverageReporters": [
       "json",
```
Comment (vincentmacri) [inline package.json:7]: If you're using lcov do you also need json?

File: .github/workflows/test.yml (hunk 1 above)
Comment (vincentmacri) [reply .github/workflows/test.yml:51]: Nevermind, saw your commit message explaining.

File: src/shared/components/__tests__/RichTable.unit.test.tsx [hunk 3]
```diff
@@ -0,0 +1,2092 @@
...
+      const { result, waitForNextUpdate } = renderHook(() =>
+        useRichTable<TestData>({
+          columns: [{ key: "id", header: "ID" }],
+          list: apiWithoutPagination,
+          paginated: false,
+        })
+      );
+      await waitForNextUpdate();
+
+      render(<div {...result.current.header.useRowProps()} />);
```
Comment (ktrieu) [inline src/shared/components/__tests__/RichTable.unit.test.tsx:920]: This test should have an `expect` in it for clarity.

File: src/shared/hooks.ts [hunk 4]
```diff
@@ -80,16 +81,16 @@ export const useSSRData: {
...
-    transformer: (resp: TResp) => TData,
+    transformer: (resp: TResp) => TData
   ): UseSSRDataHook<TData>;
   <TData, TError extends APIError = APIError>(
     dataMethod: () => Promise<APIResponse<TData, TError>>,
     initialData: TData | undefined
   ): UseSSRDataHook<TData>;
-} = <TData, TResp, TArgs, TError>(
-  dataMethod: (args?: any) => any,
+} = <TData, TResp, TError>(
```
Comment (ktrieu) [inline src/shared/hooks.ts:90]: VSCode informs me that `TResp` and `TError` aren't being used either, so we can get rid of those too.

File: src/shared/components/RichTable.tsx [hunk 5]
```diff
@@ -616,12 +646,14 @@ const useRichTable = <D extends object = {}>({
 const RichTablePagination = ({ bag }: { bag: RichTableBag<any> }) => {
   const { page, numPages, setPage, reqInfo, executeAction } = bag;
   const [newPage, setNewPage] = useState(page.toString());
-  // we need a ref to store the new page to get around the fake blur listener capturing outdated values of newPage
-  const newPageRef = useRef(page.toString());
-  const hasBlurListener = useRef(false);
+  // we need a ref to store the page to get around the fake blur listener capturing outdated values of page
+  const pageRef = useRef(page);
+  const fakeBlurListener = useRef<((e: MouseEvent) => void) | null>(null);
```
Comment (ktrieu) [inline src/shared/components/RichTable.tsx:651]: Why are we using a fake blur listener here? Whatever the reason, it should probably be documented.

Comment (tyxchen) [reply src/shared/components/RichTable.tsx]: Documentation added: 933e930

File: src/shared/hooks.ts (hunk 4 above)
Comment (tyxchen) [reply src/shared/hooks.ts]: 2009067

File: src/shared/components/__tests__/RichTable.unit.test.tsx (hunk 3 above)
Comment (tyxchen) [reply src/shared/components/__tests__/RichTable.unit.test.tsx:923]: 4b7a148

File: src/shared/components/RichTable.tsx [hunk 6]
```diff
@@ -680,51 +683,56 @@ const RichTablePagination = ({ bag }: { bag: RichTableBag<any> }) => {
...
-              e.stopPropagation();
-            }}
-            onFocus={({
-              currentTarget,
-            }: React.FocusEvent<HTMLInputElement>) => {
-              currentTarget.select();
+            onFocus={() => {
+              inputRef.current?.select();
 
               // only submit if blur was the result of a click outside the control
```
[the remark this answers is not in the document]
Comment (tyxchen) [reply src/shared/components/RichTable.tsx]: 933e930

File: package.json (hunk 2 above)
Comment (tyxchen) [reply package.json]: Good catch, I guess not. 9b6063c
